### ISSUE TITLE
Line could not be parsed error when opening a file which includes wxWidgets

### ISSUE THREAD

**Issue by MonJamp** [jbenden/vscode-c-cpp-flylint#83]
## Bug description
Using clang when parsing a file which includes the wxWidgets header `"wx/wx.h"`, I get an error which says `Line could not be parsed`. This causes the lint tool to be stopped and any warnings in code disappears.
Here is the full error that is reported:
```
vscode-c-cpp-flylint: 'Line could not be parsed: _WX_DECLARE_HASHTABLE'' while validating: /home/filjo/Documents/Code/MonPad/MonPad.cpp. Please analyze the 'C/C++ FlyLint' Output console. Stacktrace: Error: Line could not be parsed: _WX_DECLARE_HASHTABLE'
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/linters/linter.js:222:31
    at Array.forEach (<anonymous>)
    at Clang.parseLines (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/linters/linter.js:206:15)
    at Clang.lint (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/linters/linter.js:196:21)
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/server.js:262:43
    at Array.forEach (<anonymous>)
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/server.js:259:21
    at Generator.next (<anonymous>)
    at fulfilled (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.4.4/server/server.js:5:58)
```
In the log output there are many warnings (although this is kinda normal with wxWidgets) but I don't see any errors. Last line after clang lint is done analyzing is `... 12744 more items`. So there might be an error somewhere but I don't know how to find the full log.

## Steps to reproduce
1. Create a file that includes `"wx/wx.h" (make sure wxWidgets is installed)
2. Add relevant include paths and defines. In Fedora that looks like:
```
"c-cpp-flylint.clang.includePaths": [
			"/usr/include/wx-3.0/",
			"/usr/lib64/wx/include/gtk3-unicode-3.0/"
		],
"c-cpp-flylint.clang.defines": [
			"__WXGTK__",
			"WXUSINGDLL",
			"_FILE_OFFSET_BITS=64"
		],
```
3. Open file that includes the wxWidgets header

## Additional Notes
I'm pretty sure this is a bug with this extension. Clang should be able to parse wxWidgets source just fine also in the log it doesn't say that clang returned with null or any other error message. If this is a Clang error it would be nice if the extension could return the exact Clang error to make it easier to debug.

**Comment by jbenden** [comment]
Hello,

Thanks for the bug report!

This is not an error from Clang, but from the extensions use of Clang. I have a commit coming that will resolve this matter.

I'd be interested in seeing the full log from the extension on this, though. The reason is because I suspect something else is also going on... The log may be found under the "Output" panel (Ctrl-K Ctrl-H). In this panel, there is a drop-down that the "C/C++ Flylint" option may be found. Please paste/post all output.

Thanks!
-Joe

**Comment by jbenden** [comment]
Sorry, but I forgot to mention. Please enable `c-cpp-flylint.debug: true` before capturing the logs...

Thanks!
-Joe

**Comment by MonJamp** [comment]
I updated the extension and it seems that the error message has changed:
```
vscode-c-cpp-flylint: 'Line could not be parsed: note: (skipping 35 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)' while validating: /home/filjo/Documents/Code/MonPad/MonPad.cpp. Please analyze the 'C/C++ FlyLint' Output console. Stacktrace: Error: Line could not be parsed: note: (skipping 35 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:39:75700
    at Array.forEach (<anonymous>)
    at s.parseLines (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:39:75434)
    at s.lint (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:39:75337)
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:138:6706
    at Array.forEach (<anonymous>)
    at /home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:138:6662
    at Generator.next (<anonymous>)
    at s (/home/filjo/.vscode/extensions/jbenden.c-cpp-flylint-1.5.0/dist/server/server.bundle.js:138:2924)
```

And here is the output:
```
Performing lint scan of /home/filjo/Documents/Code/MonPad/MonPad.cpp...
executing:  /usr/lib64/ccache/clang -fsyntax-only -fno-color-diagnostics -fno-caret-diagnostics -fno-diagnostics-show-option -fdiagnostics-show-category=name -ferror-limit=200 -iquote /home/filjo/Documents/Code/MonPad -iquote /usr/include/wx-3.0/ -iquote /usr/lib64/wx/include/gtk3-unicode-3.0/ --std=gnu++11 -fblocks -Wall -Wextra -Weverything -D__WXGTK__ -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -I /usr/include/wx-3.0/ -I /usr/lib64/wx/include/gtk3-unicode-3.0/ -x c++ -pthread /tmp/tmp-419478zPQehJZPDn.tmp
[ '' ]
[
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  '/usr/include/wx-3.0/wx/wx.h:12:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  '/usr/include/wx-3.0/wx/defs.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  '/usr/include/wx-3.0/wx/platform.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:173:',
  '/usr/include/wx-3.0/wx/compiler.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:183:',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:5:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:61:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:67:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:70:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:73:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:89:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:97:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:855:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  '/usr/include/wx-3.0/wx/platform.h:242:16: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/platform.h:316:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:596:',
  '/usr/include/wx-3.0/wx/chkconf.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  '/usr/include/wx-3.0/wx/defs.h:54:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:193:',
  '/usr/include/wx-3.0/wx/version.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:193:',
  'In file included from /usr/include/wx-3.0/wx/version.h:16:',
  '/usr/include/wx-3.0/wx/cpp.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:120:17: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:162:30: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:164:66: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:168:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:169:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:170:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:171:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:172:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:173:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:174:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:179:31: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:570:',
  '/usr/include/wx-3.0/wx/dlimpexp.h:16:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:11:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  'In file included from /usr/include/wx-3.0/wx/debug.h:19:',
  '/usr/include/wx-3.0/wx/chartype.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:284:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:292:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:296:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:300:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:55:17: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/debug.h:145:53: warning: zero as null pointer constant [Nullability Issue]',
  "/usr/lib64/clang/10.0.0/include/stddef.h:84:18: note: expanded from macro 'NULL'",
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:205:61: warning: zero as null pointer constant [Nullability Issue]',
  "/usr/lib64/clang/10.0.0/include/stddef.h:84:18: note: expanded from macro 'NULL'",
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  "/usr/include/wx-3.0/wx/defs.h:1204:13: warning: 'long long' is incompatible with C++98",
  "/usr/include/wx-3.0/wx/defs.h:1174:26: note: expanded from macro 'wxLongLong_t'",
  "/usr/include/wx-3.0/wx/defs.h:1205:13: warning: 'long long' is incompatible with C++98",
  "/usr/include/wx-3.0/wx/defs.h:1186:36: note: expanded from macro 'wxULongLong_t'",
  "/usr/include/wx-3.0/wx/defs.h:1174:26: note: expanded from macro 'wxLongLong_t'",
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:2375:',
  '/usr/include/wx-3.0/wx/windowid.h:11:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-419478zPQehJZPDn.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  ... 12630 more items
]
Completed lint scans...
Performing lint scan of /home/filjo/Documents/Code/MonPad/MonPad.cpp...
executing:  /usr/lib64/ccache/clang -fsyntax-only -fno-color-diagnostics -fno-caret-diagnostics -fno-diagnostics-show-option -fdiagnostics-show-category=name -ferror-limit=200 -iquote /home/filjo/Documents/Code/MonPad -iquote /usr/include/wx-3.0/ -iquote /usr/lib64/wx/include/gtk3-unicode-3.0/ --std=gnu++11 -fblocks -Wall -Wextra -Weverything -D__WXGTK__ -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -I /usr/include/wx-3.0/ -I /usr/lib64/wx/include/gtk3-unicode-3.0/ -x c++ -pthread /tmp/tmp-41941dJZMYEsBWvb.tmp
[ '' ]
[
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  '/usr/include/wx-3.0/wx/wx.h:12:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  '/usr/include/wx-3.0/wx/defs.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  '/usr/include/wx-3.0/wx/platform.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:173:',
  '/usr/include/wx-3.0/wx/compiler.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:183:',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:5:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:61:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:67:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:70:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:73:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:89:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:97:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/lib64/wx/include/gtk3-unicode-3.0/wx/setup.h:855:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  '/usr/include/wx-3.0/wx/platform.h:242:16: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/platform.h:316:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:27:',
  'In file included from /usr/include/wx-3.0/wx/platform.h:596:',
  '/usr/include/wx-3.0/wx/chkconf.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  '/usr/include/wx-3.0/wx/defs.h:54:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:193:',
  '/usr/include/wx-3.0/wx/version.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:193:',
  'In file included from /usr/include/wx-3.0/wx/version.h:16:',
  '/usr/include/wx-3.0/wx/cpp.h:13:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:120:17: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:162:30: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:164:66: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:168:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:169:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:170:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:171:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:172:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:173:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:174:36: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/cpp.h:179:31: warning: variadic macros are incompatible with C++98 [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:570:',
  '/usr/include/wx-3.0/wx/dlimpexp.h:16:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:11:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  'In file included from /usr/include/wx-3.0/wx/debug.h:19:',
  '/usr/include/wx-3.0/wx/chartype.h:14:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:284:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:292:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:296:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/chartype.h:300:13: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:55:17: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  '/usr/include/wx-3.0/wx/debug.h:145:53: warning: zero as null pointer constant [Nullability Issue]',
  "/usr/lib64/clang/10.0.0/include/stddef.h:84:18: note: expanded from macro 'NULL'",
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:734:',
  '/usr/include/wx-3.0/wx/debug.h:205:61: warning: zero as null pointer constant [Nullability Issue]',
  "/usr/lib64/clang/10.0.0/include/stddef.h:84:18: note: expanded from macro 'NULL'",
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  "/usr/include/wx-3.0/wx/defs.h:1204:13: warning: 'long long' is incompatible with C++98",
  "/usr/include/wx-3.0/wx/defs.h:1174:26: note: expanded from macro 'wxLongLong_t'",
  "/usr/include/wx-3.0/wx/defs.h:1205:13: warning: 'long long' is incompatible with C++98",
  "/usr/include/wx-3.0/wx/defs.h:1186:36: note: expanded from macro 'wxULongLong_t'",
  "/usr/include/wx-3.0/wx/defs.h:1174:26: note: expanded from macro 'wxLongLong_t'",
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  'In file included from /usr/include/wx-3.0/wx/defs.h:2375:',
  '/usr/include/wx-3.0/wx/windowid.h:11:9: warning: macro name is a reserved identifier [Lexical or Preprocessor Issue]',
  'In file included from /tmp/tmp-41941dJZMYEsBWvb.tmp:1:',
  'In file included from /usr/include/wx-3.0/wx/wx.h:14:',
  ... 12630 more items
]
Completed lint scans...
```
As you can see it seems many of the warnings are cut off so there may be errors in the list but I'm unable to see them. Also it seems that the Clang linter is run twice producing the same error message and output.
Due to the error I need to enable `Ignore Parse Errors` in order for the extension to work properly.
Thanks for you hard work, I'm working on a somewhat big project and your extension is very helpful.

**Comment by jbenden** [comment]
Thank you for the logs.

Unfortunately, looks like wxWidgets is a pain mess... :)

You could try the setting below and disabling the ignore parse errors:

```
    "c-cpp-flylint.clang.warnings": [
        "all",
        "extra"
    ]
```

**Comment by MonJamp** [comment]
That works a lot better than enabling `Ignore Parse Errors`, I don't get any extension errors with just enabling those warnings.
